generic3g/tests/Test_FixedLevelsVerticalGrid.pf

```diff
@@ -13,9 +13,37 @@ contains
 
       real, parameter :: levels(*) = [1.,5.,7.]
 
-      vgrid = FixedLevelsVerticalGrid(standard_name='air_pressure', levels=levels)
+      vgrid = FixedLevelsVerticalGrid(standard_name='air_pressure', units='Pa', levels=levels)
       @assert_that(vgrid%get_num_levels(), is(size(levels)))
 
    end subroutine test_num_levels
+
+   @test
+   subroutine test_equals()
+      type(FixedLevelsVerticalGrid) :: vgrid1, vgrid2
+
+      real, parameter :: levels(*) = [1.,5.,7.]
+
+      vgrid1 = FixedLevelsVerticalGrid(standard_name='air_pressure', units='Pa', levels=levels)
+      vgrid2 = FixedLevelsVerticalGrid(standard_name='air_pressure', units='Pa', levels=levels)
+      @assert_that(vgrid1==vgrid2, is(.true.))
+
+   end subroutine test_equals
+   
+   @test
+   subroutine test_not_equals()
+      type(FixedLevelsVerticalGrid) :: vgrid1, vgrid2, vgrid3
+
+      real, parameter :: levels1(*) = [1.,5.,7.]
+      real, parameter :: levels2(*) = [.01,4.]
+
+      vgrid1 = FixedLevelsVerticalGrid(standard_name='air_pressure', units='Pa', levels=levels1)
+      vgrid2 = FixedLevelsVerticalGrid(standard_name='air_pressure', units='mb', levels=levels1)
+      vgrid3 = FixedLevelsVerticalGrid(standard_name='air_pressure', units='Pa', levels=levels2)
+      @assert_that(vgrid1 /= vgrid2, is(.true.))
+      @assert_that(vgrid1 /= vgrid3, is(.true.))
```

Comment:
I don't think this works.   You probably need to change this to:
```suggestion
      @assert_that(vgrid1 /= vgrid2, is(true()))
      @assert_that(vgrid1 /= vgrid3, is(true()))
```
I can't remember the details at the moment, but there is some reason why the annotation fails (maybe because it uses `==` under the hood?).  Might work with some compilers, but 99% certain it fails with NAG.